plutarch-test/plutarch-base/Plutarch/ListSpec.hs

```diff
@@ -1,103 +1,83 @@
+        "gt5" @| pfilter # (plam $ \x -> 5 #< x) # xs10 #== integerList [6 .. 10] @-> passert
+      "pzipWith" @\ do
+        "double" @| pzipWith' (+) # xs10 # xs10 #== integerList (fmap (* 2) [1 .. 10]) @-> passert
       "pfoldl" @\ do
-        "primed" @\ do
-          "nonempty" @| pfoldl' (-) # 0 # xs10
-            @== pconstant @PInteger (foldl (-) 0 [1 .. 10])
-          "empty" @| pfoldl' (-) # 0 # integerList []
-            @== pconstant @PInteger 0
-        "primed" @\ do
-          "nonempty" @| pfoldl # plam (-) # 0 # xs10
-            @== pconstant @PInteger (foldl (-) 0 [1 .. 10])
-          "empty" @| pfoldl # plam (-) # 0 # integerList []
-            @== pconstant @PInteger 0
-    -- Benchmarking different ways of using lists
-    describe "bench" . pgoldenSpec $ do
-      let numList = pconstant @(PBuiltinList PInteger) [1 .. 5]
-      -- Two ways of matching on a list
-      "x1+x2" @\ do
-        -- Via HeadList and TailList only.
-        "builtin" @| (phead #$ ptail # numList) + (phead # numList)
-        -- Via ChooseList (twice invoked)
-        "pmatch"
-          @| pmatch numList
-          $ \case
-            PNil -> perror
-            PCons x xs ->
-              pmatch xs $ \case
-                PNil -> perror
-                PCons y _ ->
-                  x + y
-      -- Various ways of uncons'ing a list
-      "uncons" @\ do
-        -- ChooseList builtin, like uncons but fails on null lists
-        "ChooseList"
-          @| pmatch numList
-          $ \case
-            PNil -> perror
-            PCons _x xs ->
-              xs
-        -- Retrieving head and tail of a list
-        "head-and-tail"
-          @| plet (phead # numList)
-          $ \_x ->
-            ptail # numList
-        -- Retrieve head and tail using builtins, but fail on null lists.
-        "head-and-tail-and-null"
-          @| plet (pnull # numList)
-          $ \isEmpty ->
-            pmatch isEmpty $ \case
-              PTrue -> perror
-              PFalse -> plet (phead # numList) $ \_x ->
-                ptail # numList
+        "nonempty0" @| pfoldl # plam (-) # 0 # xs10 #== pconstant (foldl (-) 0 [1 .. 10]) @-> passert
+        "nonempty1" @| pfoldl' (-) # 0 # xs10 #== pconstant (foldl (-) 0 [1 .. 10]) @-> passert
+        "empty0" @| pfoldl # plam (-) # 0 # integerList [] #== pconstant 0 @-> passert
+        "empty1" @| pfoldl' (-) # 0 # integerList [] #== pconstant 0 @-> passert
+      "elemAt" @\ do
+        "elemAt_3_[1..10]" @| pelemAt # 3 # marshal [1 .. 10 :: Integer]
+        "elemAt_0_[1..10]" @| pelemAt # 0 # marshal [1 .. 10 :: Integer]
+        "elemAt_9_[1..10]" @| pelemAt # 9 # marshal [1 .. 10 :: Integer]
+      "find" @\ do
+        "find_(==3)_[1..4]" @| pfind # plam (#== 3) #$ marshal [1 .. 4 :: Integer]
+        "find_(==5)_[1..4]" @| pfind # plam (#== 5) #$ marshal [1 .. 4 :: Integer]
```

Comment:
I'd use `integerList` instead of `marshal` since that is what this module already uses to construct lists. Or you can change everything else to use `marshal`.